docs/articles/documentation/test-api/intercepting-and-mocking-http-requests.md

```diff
@@ -0,0 +1,547 @@
+`method`    | String | The request method.
+`isAjax`    | Boolean | Specifies whether this is an AJAX request.
+`headers`   | Object | The request headers in the property-value form.
+`body`      | String | A stringified request body.
+`testRunId` | String | TestCafe internal ID of the test run that sent the request.
+
+## Attaching Hooks to Tests and Fixtures
+
+To attach a hook to a test or fixture, use the `fixture.requestHooks` and `test.requestHooks` methods. A hook attached to a fixture will handle requests from all tests in the fixture.
+
+```text
+fixture.requestHooks(...hook)
+test.requestHooks(...hook)
+```
+
+Parameter | Type | Description
+--------- | ---- | ------------
+`hook`    | RequestHook | A request logger, mock or custom hook.
+
+The `requestHooks` methods use the rest operator, which allows you to pass multiple hooks as parameters or arrays of hooks.
+
+```js
+import { RequestLogger, RequestMock } from 'testcafe';
+
+const logger = RequestLogger('http://example.com');
+const mock   = RequestMock()
+    .onRequestTo('http://external-service.com/api/')
+    .respond({ data: 'value' });
+
+fixture `My fixture`
+    .page('http://example.com')
+    .requestHooks(logger);
+
+test
+    .requestHooks(mock)
+    ('My test', async t => {
+        // test actions
+    })
+```
+
+You can also attach and detach hooks during test run. To do this, use the `t.addRequestHooks` and `t.removeRequestHooks` methods.
+
+```text
+t.addRequestHooks(...hook)
+t.removeRequestHooks(...hooks)
+```
+
+Parameter | Type | Description
+--------- | ---- | ------------
+`hook`    | RequestHook | A request logger, mock or custom hook.
+
+The `t.addRequestHooks` and `t.removeRequestHooks` methods use the rest operator, which allows you to pass multiple hooks as parameters or arrays of hooks.
+
+```js
+import { RequestLogger } from 'testcafe';
+
+const logger = RequestLogger('http://example.com');
+
+fixture `My fixture`
+    .page('http://example.com');
+
+test('My test', async t => {
+    await t
+        .click('#send-unlogged-request')
+        .addRequestHooks(logger)
+        .click('#send-logged-request')
+        .expect(logger.count(() => true)).eql(1)
+        .removeRequestHooks(logger)
+        .click('#send-unlogged-request');
+})
+```
+
+## Creating a Custom HTTP Request Hook
+
+To handle HTTP requests in a custom way, you can create your own request hook. But first, you need to know how TestCafe request hooks work.
+
+### Understanding How TestCafe Request Hooks Operate
+
+* All TestCafe request hooks descend from the `RequestHook` class.
+
+    ```js
+    class MyRequestHook extends RequestHook {
+        // ...
+    }
+    ```
+
+* To determine which requests the hook handles, the base class constructor receives an array of [filtering rules](#specifying-which-requests-are-handled-by-the-hook) as the first parameter. If no rules are passed, all requests are handled.
+
+    ```js
+    export default class RequestHook {
```

Comment:
`export default` is an extra noise for this example, just `class RequestHook...` Also you forgot to close `}`